### ISSUE TITLE
Typescript: FastifyJWTOptions.secret function type missing

### ISSUE THREAD

**Issue by AleF83** [fastify/fastify-jwt#101]
## 🐛 Bug Report

In documentation said that secret `The secret can be a primitive type String, a function that returns a String or an object { private, public }`

But in type definition file the secret type is `jwt.Secret | { public: jwt.Secret; private: jwt.Secret }`

## To Reproduce

- build the following TS code

```ts
const app = fastify()
    .register(jwtPlugin, { secret: () => 'secret' })
```

I got following error:

```bash
Type '() => string' is not assignable to type 'string | Buffer | { key: string | Buffer; passphrase: string; } | { public: Secret; private: Secret; }'.
```

## Expected behaviour

There is no build errors

## Your Environment

- *node version*: 10, 12, 13, 14
- *fastify version*: >=2.0.0
- *os*: Mac, Windows, Linux
- *any other relevant information*

**Comment by mcollina** [comment]
Would you like to send a Pull Request to address this issue? Remember to add unit tests.

**Comment by AleF83** [comment]
I'm using this module in project with fastify v2. (I'm using v1.4.0).
Can we somehow release this fix as v1.4.1 too?

**Comment by mcollina** [comment]
Please send a PR to https://github.com/fastify/fastify-jwt/tree/1.x as well. The types are different enough to make backporting hard.
